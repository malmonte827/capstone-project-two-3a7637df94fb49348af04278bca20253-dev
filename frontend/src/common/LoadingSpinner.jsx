import Spinner from 'react-bootstrap/Spinner';
import "./LoadingSpinner.css"

/**LoadingSpinner component
 * 
 * Shows a spinner loader when content is loading
 */


function LoadingSpinner() {
  return (
    <Spinner animation="border" role="status">
      <span className="visually-hidden">Loading...</span>
    </Spinner>
  );
}

export default LoadingSpinner;