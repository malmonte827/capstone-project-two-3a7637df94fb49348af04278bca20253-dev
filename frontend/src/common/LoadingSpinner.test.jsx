import React from "react";
import LoadingSpinner from "./LoadingSpinner";
import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";

describe("<LoadingSpinner />", () => {
    it("renders without crashing", () => {
        render(<LoadingSpinner />);
    });

    it("matches snapshot", () => {
        const { asFragment } = render(<LoadingSpinner />);
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders the spinner", () => {
        render(<LoadingSpinner />);

        const spinner = screen.getByRole("status");
        expect(spinner).toBeInTheDocument();

        expect(spinner).toHaveClass("spinner-border");
    });

    it("has a visually hidden loading text", () => {
        render(<LoadingSpinner />);

        const hiddenText = screen.getByText(/loading.../i);
        expect(hiddenText).toHaveClass("visually-hidden");
    });
});
