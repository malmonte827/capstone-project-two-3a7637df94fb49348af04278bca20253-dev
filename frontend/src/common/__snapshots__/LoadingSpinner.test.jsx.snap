// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<LoadingSpinner /> > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="spinner-border"
    role="status"
  >
    <span
      class="visually-hidden"
    >
      Loading...
    </span>
  </div>
</DocumentFragment>
`;
